internal/store/build_reason.go

```diff
@@ -0,0 +1,24 @@
+package store
+
+type BuildReason int
+
+const BuildReasonNone = BuildReason(0)
+
+const (
+	BuildReasonFlagFiles BuildReason = 1 << iota
+	BuildReasonFlagConfig
+	BuildReasonFlagCrash
```

Comment:
not necessary, but i'd like this to be a lil more nuanced -- we're not rebuilding b/c the pod crashed, we're rebuilding bc a pod crash may have caused us to lose state...?

there should at least be a comment.

Reply:
added a pointer to the docs on engine state about it